modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/DisasterRecoveryManager.java

```diff
@@ -277,7 +280,65 @@ private CompletableFuture<Map<TablePartitionId, Map<String, LocalPartitionStateM
             });
         }
 
-        return CompletableFuture.allOf(futures).handle((unused, throwable) -> result);
+        if (!partitionIds.isEmpty()) {
+            return allOf(futures).thenApply(ignored -> {
+                Set<Integer> foundPartitionIds = result.keySet().stream()
+                        .map(TablePartitionId::partitionId)
+                        .collect(toSet());
+
+                checkPartitions(foundPartitionIds, partitionIds);
+
+                return result;
+            });
+        }
+
+        return allOf(futures).handle((unused, throwable) -> result);
```

Comment:
I wonder why you are swallowing an exception here in the branch on line 284, no, why?

Reply:
Fixed, now throwing it as DisasterRecoveryException (it was not handled before my changes, so created a new one). On line 284 I ignore the result of allOf (it is void)